packages/sync/src/Functions.php

```diff
@@ -1,43 +1,65 @@
 <?php
+/**
+ * Utility functions to generate data synced to wpcom
+ *
+ * @package automattic/jetpack-sync
+ */
 
 namespace Automattic\Jetpack\Sync;
 
 use Automattic\Jetpack\Constants;
-/*
+
+/**
  * Utility functions to generate data synced to wpcom
  */
-
 class Functions {
 	const HTTPS_CHECK_OPTION_PREFIX = 'jetpack_sync_https_history_';
 	const HTTPS_CHECK_HISTORY       = 5;
 
+	/**
+	 * Return array of Jetpack modules.
+	 *
+	 * @return array
+	 */
 	public static function get_modules() {
 		require_once JETPACK__PLUGIN_DIR . 'class.jetpack-admin.php';
 
 		return \Jetpack_Admin::init()->get_modules();
 	}
 
+	/**
+	 * Return array of taxonomies registered on the site.
+	 *
+	 * @return array
+	 */
 	public static function get_taxonomies() {
 		global $wp_taxonomies;
 		$wp_taxonomies_without_callbacks = array();
 		foreach ( $wp_taxonomies as $taxonomy_name => $taxonomy ) {
 			$sanitized_taxonomy = self::sanitize_taxonomy( $taxonomy );
 			if ( ! empty( $sanitized_taxonomy ) ) {
 				$wp_taxonomies_without_callbacks[ $taxonomy_name ] = $sanitized_taxonomy;
-			} else {
-				error_log( 'Jetpack: Encountered a recusive taxonomy:' . $taxonomy_name );
 			}
 		}
 		return $wp_taxonomies_without_callbacks;
 	}
 
+	/**
+	 * Return array of registered shortcodes.
+	 *
+	 * @return array
+	 */
 	public static function get_shortcodes() {
 		global $shortcode_tags;
 		return array_keys( $shortcode_tags );
 	}
 
 	/**
 	 * Removes any callback data since we will not be able to process it on our side anyways.
+	 *
+	 * @param object $taxonomy WP_Taxonomy item.
```

Comment:
Instead of specifying `object`, should we specify `\WP_Taxonomy` here as it's more precise?